include/tester/ssp/SoftphoneManager.hpp

```diff
@@ -1,16 +1,98 @@
 #pragma once
 
 #include<iostream>
+#include<unordered_map>
+#include<string>
+#include<memory>
+
 #include<pjsua2.hpp>
 
 #include"Softphone.hpp"
+#include"SoftphoneArguments.hpp"
 
 class SoftphoneManager
 {
 public:
-    SoftphoneManager();
-    ~SoftphoneManager();
+    SoftphoneManager(int port, std::string domain):
+        _port(port), _domain(std::move(domain))
+    {
+    }
+
+    ~SoftphoneManager()
+    {
+        _endpoint.libDestroy();
+    }
+
+    void pjLibraryInit()
+    {
+        try
+        {
+            //Create library
+            _endpoint.libCreate();
+
+            // Init library
+            pj::EpConfig ep_cfg;
+            ep_cfg.logConfig.level = 1;
+            _endpoint.libInit( ep_cfg );
+
+            // Transport
+            pj::TransportConfig tcfg;
+            tcfg.port = _port;
+            _endpoint.transportCreate(PJSIP_TRANSPORT_UDP, tcfg);
+
+            // Start library
+            _endpoint.libStart();
+            std::cout << "*** PJSUA2 STARTED ***" << std::endl;
+        }
+        catch(pj::Error & err)
+        {
+            std::cerr << "Exception: " << err.info() << std::endl;
+        }
+    }
+
+    void runSpamTest(int amount)
+    {
+        registerSoftphones(amount * 2);
+        for(int i = 0; i < (amount * 2); i += 2)
+        {
+            if(_softphones.at(i)->isAccountValid() && _softphones.at(i + 1)->isAccountValid())
+            {
+                _softphones.at(i)->call(*_softphones.at(i + 1));
```

Comment:
Two things:
- Add an overload to the `call` method that takes in a URI and use that overload in this context.
- In `call` you should add a check if the call is inactive - only then should you be able to call.

Reply:
done